### PR TITLE
8304498: JShell does not switch to raw mode when there is no /bin/test

### PR DESCRIPTION
**Pull request by lahodaj**
If JShell is run on a system that does not have `/bin/test` (which is, apparently, possible for some systems, which only have `/usr/bin/test`), it won't switch the terminal into the raw mode, and the input will not work properly.

The proposed fix herein is to detect whether `test` existing in `/usr/bin/test`, and if yes, use that location. Use the existing `/bin/test` otherwise.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304498](https://bugs.openjdk.org/browse/JDK-8304498): JShell does not switch to raw mode when there is no /bin/test

### Reviewers
 * [Sean Coffey](https://openjdk.org/census#coffeys) (@coffeys - **Reviewer**)
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13100/head:pull/13100` \
`$ git checkout pull/13100`

Update a local copy of the PR: \
`$ git checkout pull/13100` \
`$ git pull https://git.openjdk.org/jdk.git pull/13100/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13100`

View PR using the GUI difftool: \
`$ git pr show -t 13100`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13100.diff">https://git.openjdk.org/jdk/pull/13100.diff</a>

</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
As a general point, we probably need to change jline to not create sub-processes. If we needs to introspect the environment then it will need to do it in the current VM.

File: src/jdk.internal.le/share/classes/jdk/internal/org/jline/utils/OSUtils.java [hunk 1]
```diff
@@ -102,4 +103,13 @@ public class OSUtils {
         TEST_COMMAND = test;
     }
 
+    private static boolean isTestCommandValid(String command) {
+        try {
+            Process p = new ProcessBuilder(command, "-z", "").inheritIO().start();
```
Comment (coffeys) [inline src/jdk.internal.le/share/classes/jdk/internal/org/jline/utils/OSUtils.java:108]: is there a reason why you chose to spin up a process here ? Would a test for an executable file be sufficient ?

**Review by vicente-romero-oracle — APPROVED** [review]
looks sensible

**Comment by turbanoff** [comment]
@lahodaj was the issue reported to the upstream?

**Comment by lahodaj** [comment]
> @lahodaj was the issue reported to the upstream?

I've just filled:
https://github.com/jline/jline3/issues/839

**Pull request closed without merging**